HW02/Dzmitry_Neverov/task3/time_calculator.rb

```diff
@@ -0,0 +1,50 @@
+require 'time'
+class TimeIntervalCalculator
+  attr_reader :found_rows
+
+  def initialize(found_rows)
+    @found_rows = found_rows
+  end
+
+  TIME_FORMAT = %r{/\d{4}\-\d{1,2}\-\d{1,2}
+  \s\d{1,2}\:\d{1,2}\:\d{1,2}\.\d{1,2}/}x.freeze
+
+  def search_time
+    found_rows.each { |rows| rows.match(TIME_FORMAT) }
+  end
+
+  def calc_differ(arr)
+    if (arr.length = 2)
+      difference = (arr[1] - arr[0]).to_s
+      puts(difference)
+    else
+      #   arr_for_difference = []
+      arr[1..arr.size].map!.with_index(1) do |time, i|
+        (time - arr[i - 1]).abs
+      end
+      puts("arr_for_difference #{arr}")
+    end
+  end
```

Comment:
Can you try this one?

```suggestion
  def calc_differ(arr)
      arr[1..arr.size].map!.with_index(1) do |time, i|
        (time - arr[i - 1]).abs
      end
  end
  
  puts calc_differ(arr)
```